### PR TITLE
feat: port Order.Concept

### PR DESCRIPTION
**Pull request by casavaca**
(I'm just an amateur).

In [mathlib port progress](https://tqft.net/mathlib4/2023-01-03/all.pdf), `order.concept` is a light blue node. I also checked [port_status](https://math.commelin.net/files/port_status.html), [pull request](https://github.com/leanprover-community/mathlib4/pulls?q=Order.Concept) and [Zulip](https://leanprover.zulipchat.com/#narrow/search/order.2Econcept), and it seems that no one is working on it.

So I picked it up and gave a try. With some changes I made it build.

Rename: Done.

`#lint` errors:
Edit: Resolved.

### CODE REVIEW

**Comment by casavaca** [comment]
`start_port.sh` created `import Order.Concept` but I have a feeling that it should be `import Mathlib.Order.Concept`?

**Comment by rwbarton** [comment]
> `start_port.sh` created `import Order.Concept` but I have a feeling that it should be `import Mathlib.Order.Concept`?

Definitely; I think you must have run it as `./scripts/start_port.sh Order/Concept.lean`, without the `Mathlib/` prefix. It's a bug that the script doesn't fail in this case, and manages to continue anyways.

File: Mathlib/Order/Concept.lean [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+
+
+variable (α β)
+
+/-- The formal concepts of a relation. A concept of `r : α → β → Prop` is a pair of sets `s`, `t`
+such that `s` is the set of all elements that are `r`-related to all of `t` and `t` is the set of
+all elements that are `r`-related to all of `s`. -/
+structure Concept extends Set α × Set β where
+  closure_fst : intentClosure r fst = snd
+  closure_snd : extentClosure r snd = fst
```
Comment (kbuzzard) [inline Mathlib/Order/Concept.lean:176]:
```suggestion
  /-- The axiom of a `Concept` stating that the closure of the first set is the second set. -/
  closure_fst : intentClosure r fst = snd
  /-- The axiom of a `Concept` stating that the closure of the second set is the first set. -/
  closure_snd : extentClosure r snd = fst
```
That will get rid of two of your linter errors

**Review by kbuzzard — COMMENTED** [review]
As for the second linter error, there is currently an ongoing discussion on Zulip about whether we should allow `simp` lemmas which can be proved with `simp`.

**Comment by kbuzzard** [comment]
As for the other two linter errors, just comment out `@[simp]` with `-- ` and add a porting note explaining why.

**Comment by casavaca** [comment]
Abandoned. I'll create a new one as a branch of leanprover-community/mathlib4

**Pull request closed without merging**